storm-core/src/jvm/backtype/storm/scheduler/Cluster.java

```diff
@@ -430,12 +457,44 @@ public SupervisorDetails getSupervisorById(String nodeId) {
     }
 
     /**
+     * set assignments for cluster
+     */
+    public void setAssignments(Map<String, SchedulerAssignment> newAssignments) {
+        this.assignments = new HashMap<String, SchedulerAssignmentImpl>();
+        for (Map.Entry<String, SchedulerAssignment> entry : newAssignments.entrySet()) {
```

Comment:
Looking at this snippet of code, it is not clear what the entry's key contains. We just learn that it is a String, and is put in the assignments map as a key again. This is why I like the other style of looping: 

for (String representativeNameOfTheKeyString: newAssignments.keySet()) {
      String representativeNameOfTheValueString: newAssignments.get(representativeNameOfTheKeyString);
      //do something with key and value